tests/main.rs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::datatypes::{DataType, TimeUnit};
+use utils::{display_val, run_query};
+
+mod utils;
+
+#[tokio::test]
+async fn test_mode_utf8() {
+    let sql = "SELECT MODE(utf8_col) FROM test_table";
+    let batches = run_query(sql).await.unwrap();
+    assert_eq!(display_val(batches).await, (DataType::Utf8, "apple".to_string()));
+}
+
+#[tokio::test]
+async fn test_mode_int64() {
+    let sql = "SELECT MODE(int64_col) FROM test_table";
+    let batches = run_query(sql).await.unwrap();
+    assert_eq!(display_val(batches).await, (DataType::Int64, "3".to_string()));
+}
+
+#[tokio::test]
+async fn test_mode_float64() {
+    let sql = "SELECT MODE(float64_col) FROM test_table";
+    let batches = run_query(sql).await.unwrap();
+    assert_eq!(display_val(batches).await, (DataType::Float64, "3.0".to_string()));
+}
+
+#[tokio::test]
+async fn test_mode_date64() {
+    let sql = "SELECT MODE(date64_col) FROM test_table";
+    let batches = run_query(sql).await.unwrap();
+    assert_eq!(
```

Comment:
Some potential ideas to make testing easier:
1. use [assert_batches_eq](https://docs.rs/datafusion/latest/datafusion/macro.assert_batches_eq.html)
2.  integrate https://crates.io/crates/sqllogictest (we can avoid most of the stuff in the DataFusion runner, and it could be much simpler here I think) to assist in testing
2.  Use insta.rs (my personal suggestion) to compare expected and actual (it automates the capture and update of expected output). Here is an example

https://github.com/datafusion-contrib/datafusion-dft/blob/fbb039fab11a27133b2f06857007b9095d607e5c/tests/extension_cases/functions_json.rs#L35-L54

Reply:
@alamb thanks for the review!

I’ve refactored the tests using insta, and I must say, it looks great now. 

Regarding the sqllogictest, am I correct in understanding that we won’t be reusing the setup from DataFusion core and will instead implement it in place?

Reply:
>  am I correct in understanding that we won’t be reusing the setup from DataFusion core and will instead implement it in place?



that is one proposal -- but I think you should consider if that is worth while or if Insta is ok for this repo.